src/fees/AaveFlashLoanFeeCalculator.sol

```diff
@@ -20,31 +20,17 @@ contract AaveFlashLoanFeeCalculator is IFeeCalculator, FeeCalculatorBase {
 
     function getFees(address to, bytes calldata data) external view returns (IParam.Fee[] memory) {
         // Aave flash loan signature:'flashLoan(address,address[],uint256[],uint256[],address,bytes,uint16)', selector: 0xab9c4b5d
-        (, address[] memory tokens, uint256[] memory amounts, , , bytes memory params, ) = abi.decode(
+        (, address[] memory tokens, uint256[] memory amounts, , , , ) = abi.decode(
             data[4:],
             (address, address[], uint256[], uint256[], address, bytes, uint16)
         );
 
         amounts = calculateFee(amounts);
         bytes32 metadata = to == IAaveV3Provider(aaveV3Provider).getPool()
-            ? _V3_FLASHLOAN_META_DATA
-            : _V2_FLASHLOAN_META_DATA;
+            ? _V3_FLASH_LOAN_META_DATA
+            : _V2_FLASH_LOAN_META_DATA;
 
-        IParam.Fee[] memory feesWithFlashLoan = _createFees(tokens, amounts, metadata);
-
-        if (params.length > 0) {
-            // Decode data in the flash loan
-            (IParam.Logic[] memory logics, , ) = abi.decode(params, (IParam.Logic[], IParam.Fee[], address[]));
-
-            // Get fees
-            IParam.Fee[] memory feesInFlashLoanData = Router(router).getFeesByLogics(logics, 0);
-
-            if (feesInFlashLoanData.length > 0) {
-                feesWithFlashLoan = _concatenateFees(feesWithFlashLoan, feesInFlashLoanData);
-            }
-        }
-
-        return feesWithFlashLoan;
+        return _createFees(tokens, amounts, metadata);
```

Comment:
I suggest considering the idea of calculating the fee amounts within the `_createFees()` function for improved efficiency. 
```suggestion
    function _createFees(
        address[] memory tokens,
        uint256[] memory amounts,
        bytes32 metadata
    ) internal pure returns (IParam.Fee[] memory) {
        uint256 length = tokens.length;
        IParam.Fee[] memory fees = new IParam.Fee[](length);
        for (uint256 i; i < length; ) {
            fees[i] = IParam.Fee({token: tokens[i], amount: calculateFee(amounts[i]), metadata: metadata});

            unchecked {
                ++i;
            }
        }
        return fees;
    }
```

Reply:
Refined in https://github.com/dinngo/composable-router-contract/pull/72/commits/728eec197cd00a705ade5d2e8f7433fa1b934809.

I removed `_createFees` and inline the loop to `getFees` to be consistent with all fee calculators.

Reply:
Given that `FeeCalculatorBase`'s `calculateFee(uint256[])` and `calculateAmountWithFee(uint256[])` functions are no longer in use following this change, and the likelihood of future use seems low. I propose we remove these two functions.  This would decrease the deployment cost for all fee calculators.  

I'm keen to hear your thoughts on this.

Reply:
If some functions are not needed anymore, I am okay with getting rid of them.

Reply:
Just realized that only `calculateFee(uint256[])` is unused but `calculateAmountWithFee(uint256[])` is used. I think we can put this idea to the existing asana tasks, don't fix in this PR.